### ISSUE TITLE
iOS: Unable to install reanimated

### ISSUE THREAD

**Issue by rrebase** [microsoft/react-native-test-app#1195]
### Description
Adding `"react-native-reanimated": "2.10.0"` to the example app, installing pods and running the app leads to 
```
TypeError: undefined is not an object (evaluating 'this.InnerNativeModule.installCoreFunctions')
```
even after adding the reanimated [babel plugin](https://docs.swmansion.com/react-native-reanimated/docs/fundamentals/installation#babel-plugin) and running metro with cache cleared. Looks like the JSI is not linked/installed.

Is this something expected atm? Didn't see it reported among known issues [https://github.com/microsoft/react-native-test-app/wiki/Troubleshooting](https://github.com/microsoft/react-native-test-app/wiki/Troubleshooting)

**Comment by tido64** [comment]
Hi @rrebase, what version of react-native are you using? Are you building with new arch enabled? Do you have a minimal repro I can have a look at? I don't think I've seen this issue before.

**Comment by rrebase** [comment]
```
options = {
  :hermes_enabled => true,
  :fabric_enabled => false,
  :turbomodule_enabled => false,
}
```

RN versions 0.68.2 & 0.70.0. I'll prepare a minimal repro 🙂

**Comment by rrebase** [comment]
Minimal repro https://github.com/rrebase/react-native-test-app/tree/reanimated-demo

**Comment by tido64** [comment]
@rrebase: I managed to repro it. I also found this issue: https://github.com/software-mansion/react-native-reanimated/issues/3420

There's a workaround posted here: https://github.com/software-mansion/react-native-reanimated/issues/2791#issuecomment-1004779673

I tried it and it seems to be working:

![image](https://user-images.githubusercontent.com/4123478/200338307-13f5d942-65aa-4d25-8d56-6abefdf6f4ec.png)

I'm not sure what's going on though. But it doesn't seem like it's an issue with react-native-test-app in particular. It seems like `installTurboModule()` is making some assumptions that isn't always true.

Update: I had a closer look at how `installTurboModule()` works and it's quite… hacky, to say the least. This is _not_ a reliable way to install a custom runtime at all, and it's basically polluting _all_ classes inheriting from `UIResponder`. Since it is a fairly popular package, I think we will have to support it somehow if upstream can't come up with a better solution. Definitely not like this though.
